routes/soldiers.js

```diff
@@ -0,0 +1,75 @@
+import events from 'events';
+import { connectToDB, client } from '../connections.js';
+import { addNewSoldier, lookForSoldier, lookForAllSoldiers } from '../database.js';
+
+const eventEmitter = new events.EventEmitter();
+eventEmitter.on('load', connectToDB);
+
+const logOptions = {
+  level: process.env.logLevel || 'info',
+};
+
+const newSoldierSchema = {
+  body: {
+    type: 'object',
+    required: ['_id', 'name', 'rank', 'limitations'],
+    properties: {
+      _id: { type: 'integer' },
+      name: { type: 'string' },
+      rank: { type: 'integer' },
+      limitations: { type: 'array' },
+    },
+
+  },
+};
+
+export default function router(app, opts, done) {
+  app.logLevel = logOptions;
+  app.post('/', { newSoldierSchema }, async (req, res) => {
+    const soldier = req.body;
+    try {
+      await addNewSoldier(client, {
```

Comment:
why import the client then? you just pass it... you can import it in the `database.js`. 
(You can maybe import it in the app or server, but here it is kind a meaningless)

Reply:
fixed